libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -0,0 +1,275 @@
+import {
+  AnyProviderEvent,
+  ClientProviderEvents,
+  EvaluationContext,
+  FlagMetadata,
+  FlagNotFoundError,
+  FlagValue,
+  GeneralError,
+  Hook,
+  JsonValue,
+  Logger,
+  OpenFeatureError,
+  Provider,
+  ProviderEventEmitter,
+  ProviderFatalError,
+  ResolutionDetails,
+  TypeMismatchError,
+} from '@openfeature/web-sdk';
+import { OfrepWebProviderOptions } from './model/ofrep-web-provider-options';
+import {
+  EvaluationFailureErrorCode,
+  EvaluationRequest,
+  EvaluationResponse,
+  handleEvaluationError,
+  isEvaluationFailureResponse,
+  isEvaluationSuccessResponse,
+  OFREPApi,
+  OFREPApiFetchError,
+  OFREPApiTooManyRequestsError,
+  OFREPApiUnauthorizedError,
+  OFREPForbiddenError,
+  RequestOptions,
+} from '@openfeature/ofrep-core';
+import { EvaluationStatus } from './model/evaluation-status';
+import {
+  InvalidContextError,
+  ParseError,
+  StandardResolutionReasons,
+  TargetingKeyMissingError,
+} from '@openfeature/core';
+import { isResolutionError, ResolutionError } from './model/resolutionError';
+
+export class OfrepWebProvider implements Provider {
+  DEFAULT_POLL_INTERVAL = 30000;
+
+  metadata = {
+    name: OfrepWebProvider.name,
+  };
+  readonly runsOn = 'client';
+  events?: ProviderEventEmitter<AnyProviderEvent, Record<string, unknown>> | undefined;
+  hooks?: Hook[] | undefined;
+
+  // logger is the Open Feature logger to use
+  private _logger?: Logger;
+  // _options is the options used to configure the provider.
+  private _options: OfrepWebProviderOptions;
+
+  private _ofrepAPI: OFREPApi;
+  private _etag: string | undefined;
+  private _cache: { [key: string]: ResolutionDetails<FlagValue> | ResolutionError } = {};
+  private _context: EvaluationContext | undefined;
+  private _pollingIntervalId?: number;
+
+  constructor(options: OfrepWebProviderOptions, logger?: Logger) {
+    this._options = options;
+    this._logger = logger;
+    this._ofrepAPI = new OFREPApi(this._options.baseUrl);
+  }
+
+  resolveBooleanEvaluation(flagKey: string): ResolutionDetails<boolean> {
+    return this.evaluate(flagKey, 'boolean');
+  }
+  resolveStringEvaluation(flagKey: string): ResolutionDetails<string> {
+    return this.evaluate(flagKey, 'string');
+  }
+  resolveNumberEvaluation(flagKey: string): ResolutionDetails<number> {
+    return this.evaluate(flagKey, 'number');
+  }
+  resolveObjectEvaluation<T extends JsonValue>(flagKey: string): ResolutionDetails<T> {
+    return this.evaluate(flagKey, 'object');
+  }
+
+  /**
+   * onContextChange is called when the context changes, it will re-evaluate the flags with the new context
+   * and update the cache.
+   * @param oldContext - the old context (we are not using it)
+   * @param newContext - the new context
+   */
+  async onContextChange?(oldContext: EvaluationContext, newContext: EvaluationContext): Promise<void> {
+    try {
+      if (oldContext === newContext) {
```

Comment:
This would cause the `onContextChange` method to exit early if someone did this:

```ts
const context = { name: "mike" };

await OpenFeature.setContext(context);
context.name = "thomas";
// This would be skipped
await OpenFeature.setContext(Context);
```

This behavior may be surprising to users.

Reply:
good catch, I will remove this early return.